contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -4627,4 +4639,38 @@ static Node* optimize_explicit_cast(ParseState *pstate, Node *node)
 		}
 	}
 	return node;
+}
+
+static void babelfixedparallelstate_insert(ParallelContext *pcxt, bool estimate)
+{
+	BabelfishFixedParallelState *bfps;
+	int len;
+	char* current_db_name;
+	if (estimate)
+	{
+		/* Allow space to store the babelfish fixed-size parallel state. */
+		shm_toc_estimate_chunk(&pcxt->estimator, sizeof(BabelfishFixedParallelState));
+		shm_toc_estimate_keys(&pcxt->estimator, 1);
+	}
+	else
+	{
+		/* Initialize babelfish fixed-size state in shared memory. */
```

Comment:
we rely on tds connection to call this hook. Another option is to rely on database name being NULL or not. Check can happen here as well.

Reply:
I think it is better to have the check for tds connection only, because IMO there can be other case where the db may be NULL (do not have any example in my mind right now). Can you please provide any scenario where tds connection check can cause any issue?